compiler/qsc_parse/src/expr.rs

```diff
@@ -710,9 +711,20 @@ fn lambda_op(s: &mut ParserContext, input: Expr, kind: CallableKind) -> Result<B
     Ok(Box::new(ExprKind::Lambda(kind, input, output)))
 }
 
-fn field_op(s: &mut ParserContext, lhs: Box<Expr>) -> Result<Box<ExprKind>> {
+#[allow(clippy::unnecessary_wraps)]
```

Comment:
This is from the `pedantic` profile of clippy, and I myself turn off a lot of pedantic lints too. Is it worth using the normal clippy profile?

Some other lints that have been bothering me:
- disallowing `Default::default()`
- needless raw string hashes

There are a lot of other profiles we can explore in clippy, perhaps now is the time?

Reply:
For example, this clippy lint here is actually invalid because you need the function to match a certain signature. These kinds of lints are generating more work, not less.